packages/cli/tests/integration/git.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { tmpdir } from 'node:os';
+import { mkdir, mkdtemp, rm, writeFile } from 'node:fs/promises';
+import { dirname, join } from 'node:path';
+import { execute } from '../../src/lib/executeCommand';
+import { getWorkingDiff } from '../../src/lib/git';
+
+describe('git', () => {
+  let projectDirectory: string;
+
+  const createFile = async (file: string, content: string) => {
+    const directory = dirname(file);
+    await mkdir(directory, { recursive: true });
+    await writeFile(join(projectDirectory, file), content);
+  };
+
+  const commitFile = async (file: string, message: string) => {
+    await execute('git', ['add', file], { cwd: projectDirectory });
+    await execute('git', ['commit', '-m', message], { cwd: projectDirectory });
+  };
+
+  const createBranch = (branch: string) =>
+    execute('git', ['checkout', '-b', branch], { cwd: projectDirectory });
+
+  beforeEach(async () => {
+    projectDirectory = await mkdtemp(join(tmpdir(), 'appmap-git-test-'));
```

Comment:
This package already uses `tmp` and `tmp-promise` packages which might be more convenient and structured for tmpdir creation.